lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -9484,9 +9484,8 @@ using namespace Js;
         }
         else
         {
-            Assert(fromVal < MaxArrayLength);
-            Assert(toVal < MaxArrayLength);
-            Assert(direction == -1 || (fromVal + count < MaxArrayLength && toVal + count < MaxArrayLength));
+            Assert(direction != -1 || (fromVal < MaxArrayLength && toVal < MaxArrayLength));
+            Assert(fromVal + count <= MaxArrayLength && toVal + count <= MaxArrayLength);
```

Comment:
This may be obvious, but what is going on in these asserts should be to opposite of the condition for "slower path" above (currently line 9460) which reads like "backwards and from/to large OR from+count/to+count large'. I suspect the assert that is tripping is off by one element, or maybe overflowing, not sure.

Reply:
Previously, it was not the logical opposite of the condition; now it should.

> one off
It used to say `<` instead of `<=`... 

I'm in general not sure, why `assert`s are used in such places at all; Why checking a condition, that is just checked above (`if`)??

Reply:
Or do you mean, that we may need to change the `if` condition itself?!

Reply:
Probably not, just curious why this way works. I thought the conditions in asserts are the way they because "faster path" can only handle indices within certain range.

Reply:
This actually is an off-by-one error just as you suggested!
If `copyWithin` receives no arguments for `count` it will choose `length`.
If `length == uint32.MaxValue` => assert fails cause it's not the logical complement of the `if` condition.

Reply:
Awesome, I am glad we got to the bottom of it!